src/config.ts

```diff
@@ -640,6 +641,21 @@ function validateConfigAgainstV1(rawConfig: any, cliFlags: any) {
   }
 }
 
+export function createConfiguration(config: Partial<SnowpackConfig>): SnowpackConfig {
```

Comment:
this is a good start, but the logic should be shared with `loadAndValidateConfig` to reduce behavioral differences between the two. Can you reformat so that `loadAndValidateConfig()` calls this function once it's done loading config from the file system?

Reply:
OK! I made a few changes. Now  `loadAndValidateConfig` calls  `createConfiguration` and both functions use the same validation logic. With the help of `validateOnce` function, config will not be validate twice both in  `loadAndValidateConfig` and `createConfiguration`